lib/model/query/forms.js

```diff
@@ -21,8 +21,21 @@ module.exports = {
     forms._get(extended, (q) => q.where({ xmlFormId })).then(maybeFirst),
 
   getAll: (extended) => ({ forms }) => forms._get(extended),
-  getOpen: (extended) => ({ forms, Form }) =>
-    forms._get(extended, (q) => q.whereNotIn('state', [ Form.states().closing, Form.states().closed ])),
+
+  getAllForOpenRosa: () => ({ db, Form }) =>
+    db.select(Form.fields().concat([ 'hasAttachments' ]))
+      .from('forms')
+      .whereNotIn('state', [ Form.states().closing, Form.states().closed ])
+      .leftOuterJoin(
+        db.select(db.raw('"formId", (count("formId") > 0) as "hasAttachments"'))
```

Comment:
sort of yes, sort of no.

yes, it'll always be true if it exists at all. but we still have to produce _some_ value to pipe up to the join table, so it may as well be true. the only possible values are true and null, which i don't have a big problem with.

on the other hand, no; the count cannot go on the main table. the final result of the join table inner result is a tuple (formId, hasAttachments). if we don't group by formId on the join table, we would have to do so on the outer join (which would result in a really complex query to actually project all our other columns out). if we leave the group on the join table but don't count while we are at it, the only tuple we can ever retrieve is (formId). one supposes that the presence of that value is sufficient for what we need, but it's sort of an awkward way to formulate the data: you can't reduce it to a nonredundant value (true, for instance) because you still need to be able to join on it. but you then end up with a weird form_attachments."formId" value whose presence somehow means that there are attachments on the form. yes, you could then transform that value on the main select, but it doesn't make the query any easier to understand.

Reply:
I see what you're saying, and I'm happy for it to stay as-is. Maybe something like `form_attachments."formId" is not null as "hasAttachments"` in the main select would also do the trick, but I think the current version also works.